activerecord/lib/active_record/connection_adapters/postgresql/schema_definitions.rb

```diff
@@ -45,130 +45,780 @@ def primary_key(name, type = :primary_key, **options)
         ##
         # :method: bigserial
         # :call-seq: bigserial(*names, **options)
```

Comment:
I've moved all the Data Types from the PostgreSQL guides to this API doc. I've also added more examples and added details and examples to the additional methods.